src/main/java/gregtech/integration/jei/GTOreInfo.java

```diff
@@ -0,0 +1,325 @@
+package gregtech.integration.jei;
+
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.api.unification.material.type.Material;
+import gregtech.api.worldgen.config.FillerConfigUtils;
+import gregtech.api.worldgen.config.OreDepositDefinition;
+import gregtech.api.worldgen.filler.BlockFiller;
+import gregtech.api.worldgen.filler.FillerEntry;
+import gregtech.api.worldgen.populator.FluidSpringPopulator;
+import gregtech.api.worldgen.populator.IVeinPopulator;
+import gregtech.api.worldgen.populator.SurfaceBlockPopulator;
+import gregtech.api.worldgen.populator.SurfaceRockPopulator;
+import it.unimi.dsi.fastutil.ints.IntSortedSet;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.DimensionType;
+import net.minecraft.world.biome.Biome;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fluids.*;
+
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.util.*;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+
+import static gregtech.api.GTValues.M;
+
+public class GTOreInfo implements IRecipeWrapper {
+
+    private final OreDepositDefinition definition;
+    private final int maxHeight;
+    private final int minHeight;
+    private final String name;
+    private final int weight;
+    private final int[] dimensionIDs;
+    private final IVeinPopulator veinPopulator;
+    private final BlockFiller blockFiller;
+    private List<List<ItemStack>> groupedInputsAsItemStacks = new ArrayList<>();
+    private List<List<ItemStack>> groupedOutputsAsItemStacks = new ArrayList<>();
+    private final Function<Biome, Integer> biomeFunction;
+    private Map<DimensionType, IntSortedSet> dimMap =  DimensionManager.getRegisteredDimensions();
+
+    public GTOreInfo(OreDepositDefinition definition) {
+        this.definition = definition;
+        //Some veins don't have a maximum height, so set it to the maximum world height?
+        this.maxHeight = definition.getMaximumHeight() == Integer.MAX_VALUE ? 255 : definition.getMaximumHeight();
+        //Some veins don't have a minimum height, so set it to 0 in that case
+        this.minHeight = definition.getMinimumHeight() == Integer.MIN_VALUE ? 0 : definition.getMinimumHeight();
```

Comment:
We have support for Cubic chunks so heights should not be locked by standard MC min/max

Reply:
If desired, I could add more compatibility with Cubic Chunks and get their minimum and maximum world height from their API. However, it would require adding the Cubic Chunks API to the project.

Reply:
No need for adding Cubic Chunks just don't cut Height, provide value which you get from file.

Reply:
Some files don't provide a minimum or maximum height, so the values are set to - Max Int and + Max Int respectively, would could be confusing for vanilla worlds. Instead, shall I check if Cubic Chunks is loaded in the mods list, and if it is not then use the vanilla values, and if it is just let it use - Max Int and + Max Int?

Reply:
This proposal seems reasonable, please do.

Reply:
I have now done this. The height won't get cut to vanilla minimums and maximums if cubic chunks is loaded. If cubic chunks is not loaded, it will continue being cut to vanilla values.